### PR TITLE
feat: add btc fee choice

### LINKED ISSUE #3141: Add fees screen for BTC and inscription send flows

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/4690047791).

Add btc fee choice in send flow

### CODE REVIEW

File: src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form.tsx
Comment (alter-eggo) [inline src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form.tsx:83]: @kyranjamie @fbwoolf  not sure how to do it better. in tests we need to wait until `calcMaxSpend` will provide valid result

File: src/app/pages/send/send-crypto-asset-form/components/send-crypto-asset-form.layout.tsx
Comment (fbwoolf) [inline src/app/pages/send/send-crypto-asset-form/components/send-crypto-asset-form.layout.tsx]: Does this need to be rebased? Didn't you already fix this?

**Comment by fbwoolf** [comment]
> 
> Add btc fee choice in send flow
>
Would you mind always including screenshots of the UI work you do in PRs?

File: src/app/pages/send/send-crypto-asset-form/form/btc/btc-set-fee.tsx
Comment (fbwoolf) [inline src/app/pages/send/send-crypto-asset-form/form/btc/btc-set-fee.tsx:23]: I think the naming throughout this PR needs to accommodate ordinal txs as well, so using `Bitcoin` in naming rather than `Btc` ...maybe to match nav `toChooseBitcoinFee` could be `BitcoinChooseFee` (and route to match)? Also, I think this needs to be located in the `Family` folder and reused for ordinal sends (see attached issue)? It might be that we need to move that form first to make sharing components b/w these forms work?

**Comment by fbwoolf** [comment]
@alter-eggo this looks really good, but I think there is more work to add the fee selection to the ordinal flow too ...which, I think, might require refactoring/relocating that form code?

File: src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form.tsx
Comment (fbwoolf) [reply src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form.tsx:83]: Thinking on this, it doesn't work just to add a `waitFor` in the test? It looks like the sendMaxButton is only used in one test. Can you wait for a value in the amount input in the test itself?

Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form.tsx:83]: >  Can you wait for a value in the amount input in the test itself

don't understand what you mean, how it will help here? 🤔

File: src/app/pages/send/send-crypto-asset-form/components/send-crypto-asset-form.layout.tsx
Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/components/send-crypto-asset-form.layout.tsx]: yeah, already fixed

File: src/app/pages/send/send-crypto-asset-form/form/btc/btc-set-fee.tsx
Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/form/btc/btc-set-fee.tsx:23]: yeah, agree here

File: src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form.tsx
Comment (fbwoolf) [reply src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form.tsx:83]: I think I confused this with the STX form not realizing what was happening with the `calcMaxSpend` for btc. Not sure I have a better solution than the one you are proposing. 🤔

**Comment by alter-eggo** [comment]
btc flow:
https://user-images.githubusercontent.com/46521087/228943117-5788fee5-5a62-4080-8b22-68021787ace5.mov

inscription flow (here I temporary removed validation locally to show flow):
https://user-images.githubusercontent.com/46521087/228943187-fb1e96d6-9cf7-4deb-b77b-7095aa170501.mov

**Comment by markmhendrickson** [comment]
Looking great!

**Comment by markmhendrickson** [comment]
Noticing this error message displays briefly upon transition?

Screenshot 2023-03-31 at 18 53 07

**Comment by alter-eggo** [comment]
@markmhx yeah, I did it purposefully to show the flow in the video, normally this error doesn't allow to continue the send flow

**Comment by alter-eggo** [comment]
@fbwoolf @kyranjamie so we can merge this as is, implementing only btc set fees, and when https://github.com/hirosystems/wallet/issues/3399 is ready, I will add inscription fees choice in separate pr

**Review by fbwoolf — APPROVED** [review]

File: src/app/pages/send/send-crypto-asset-form/family/bitcoin/components/useBitcoinSetFees.ts
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/family/bitcoin/components/useBitcoinSetFees.ts]: rename file with snake casing

File: src/app/pages/send/send-crypto-asset-form/components/fees-card.tsx
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/components/fees-card.tsx]: Can we do this in css with `_hover={{ background: '#F9F9FA' }}`

File: src/app/pages/send/send-crypto-asset-form/family/bitcoin/components/useBitcoinSetFees.ts
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/family/bitcoin/components/useBitcoinSetFees.ts]: This logic assumes that the transaction is going to contain all uninscribed taproot UTXOs, which is likely not the case.

For calculating inscription transfer fees, we'll use TR utxos (though for now, just `1`).
For calculating btc transfers, the number of inputs is dependent on the amount being transferred. 

This is where it gets tricky. It could be that for a transfer of _x_ the low fee estimate, might use 3 UTXOs, and that a high fee estimate uses 5.

For this case, I think we'll need to run the coinselection in `determineUtxosForSpend` 3 times, with varying fee rates, to determine the selections.

**Review by kyranjamie — COMMENTED** [review]
Great work Anastas, sorry for the slow review.

Happy to jump on a call to help chat through this implementation if you like.

File: src/app/pages/send/send-crypto-asset-form/family/bitcoin/components/use-bitcoin-set-fees.ts
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/family/bitcoin/components/use-bitcoin-set-fees.ts]:
```suggestion
interface UseBitcoinSetFeesArgs {
```

pascal case naming

**Review by kyranjamie — APPROVED** [review]
👌🏼

**Comment by mica000** [comment]
This is looking good @alter-eggo!

Wondering about the location of the footer. Is there a reason the bottom is bigger? 
Screenshot 2023-04-11 at 11 40 53



Ideally, this should work like this:
Screenshot 2023-04-11 at 11 44 54

**Comment by alter-eggo** [comment]
@mica000 not sure here, I'll fix

**Comment by alter-eggo** [comment]
hm, there is hardcoded 106px height https://github.com/hirosystems/wallet/blob/58520fd2a432d1c1c91648eca2e9d4074108bbe1/src/app/pages/send/send-crypto-asset-form/components/form-footer.tsx#L20

@fbwoolf was there any specific reason to set it to 106?

**Comment by mica000** [comment]
@alter-eggo Are we not informing about this anymore?
"Fees are deducted from your balance, it won’t affect your sending amount."

Screenshot 2023-04-11 at 13 24 31

**Comment by mica000** [comment]
@alter-eggo 

Same applies to this footer as well although think we agreed before on updating all footers to be fixed on the bottom:
Screenshot 2023-04-11 at 11 41 42

---

More like this, in order to cover for cases like STX when the content is quite big
Screenshot 2023-04-11 at 13 32 38
Screenshot 2023-04-11 at 13 33 57

**Comment by alter-eggo** [comment]
@mica000 
> @alter-eggo Are we not informing about this anymore?

I added this today, you prob need to download new build

**Comment by alter-eggo** [comment]
@mica000  if we fix footer to bottom, user won't see the memo warning, until scroll down, which I think most of the users won't do, are we ok with it?

Screenshot 2023-04-11 at 16 28 01

**Comment by mica000** [comment]
> @mica000 if we fix footer to bottom, user won't see the memo warning, until scroll down, which I think most of the users won't do, are we ok with it?
> 
> Screenshot 2023-04-11 at 16 28 01

Good point! For the popover, we could move the notification to the top? wdyt?
Screenshot 2023-04-11 at 14 52 48

File: src/app/components/info-card/info-card.tsx
Comment (kyranjamie) [inline src/app/components/info-card/info-card.tsx]: Just a note that, if we ever want this component without the `px` it becomes as hassle. Better to add it separately on each use?

**Review by kyranjamie — APPROVED** [review]

File: src/app/components/info-card/info-card.tsx
Comment (alter-eggo) [reply src/app/components/info-card/info-card.tsx]: yeah, agree here, done

**Comment by alter-eggo** [comment]
@mica000 I made all the ui fixes

**Comment by mica000** [comment]
@alter-eggo thanks! 

Spotted a small copy error: 
**Fees** are deducted from your balance, it won't affect your sending amount.

Currently, it seems the copy is left-aligned, could you also align this copy to center?

**Comment by mica000** [comment]
@alter-eggo the footer error seems to persist on the popover when sending BTC. For STX works fine tho.

Screenshot 2023-04-12 at 12 47 02
Screenshot 2023-04-12 at 12 47 20

**Comment by mica000** [comment]
@alter-eggo when using the fullscreen view, the corners are gone:

Screenshot 2023-04-12 at 12 45 14
Screenshot 2023-04-12 at 12 45 31

**Comment by alter-eggo** [comment]
@mica000 fixed

**Comment by fbwoolf** [comment]
Is this one ready to merge/release yet?

EDIT: I believe I am still waiting for it to create the release?

**Comment by alter-eggo** [comment]
I think yes, it's ready

**Comment by fbwoolf** [comment]
> I think yes, it's ready

I'll let you merge it when ready.

**Comment by mica000** [comment]
@alter-eggo Almost done, but found this happening on the footer of the fullpage when sending STX:

It seems like some containers seem to have a fixed height and others are flexible?

Screenshot 2023-04-13 at 15 26 13


Other than that, its looking great!

**Comment by alter-eggo** [comment]
@mica000 it happens bc I added padding-bottom to page as in popup mode content is hidden beside fixed bottom btn. 
fixed in full page mode, thanks a lot for all the reviews 🙌

**Pull request merged**
